embedded/sql/stmt.go

```diff
@@ -518,33 +533,57 @@ func (stmt *UpsertIntoStmt) execAt(tx *SQLTx, params map[string]interface{}) (*S
 			}
 
 			if rval.IsNull() {
-				if col.notNull {
+				if col.notNull || col.autoIncrement {
```

Comment:
if value provided is null and col is autoincrement why we raise an error? It should be autogenerated...

Reply:
it shouldn't be explicitly specified as NULL, I think it could make issues harder to find if a value is autogenerated when null value is explicitly set... but we could check how other databases behave on this...

Reply:
checked in pgsql it behaves the same manner